src/embedded.js

```diff
@@ -606,11 +609,18 @@
                 window.addEventListener('scroll', this.fixIframe);
             }
 
+            // Close the iframe if page fails to initialize within 15 seconds
+            var iframeHealthCheck = setTimeout(function() {
+                HelloSign.reportError('Signer page failed to initialize within 15 seconds.', document.location.href);
+                HelloSign.close();
+            }, HelloSign.healthCheckInterval);
```

Comment:
@freddyrangel I think you're going to have to store the handle within the `HelloSign` object scope. The `XWM.receive(function() { ... })` scope below won't have access to the handle. Also, can you name this something more descriptive like `HelloSign._healthCheckTimeoutHandle` (underscore not necessary, but could be good to indicate this is a private/internal property - up to you)?